rust/otap-dataflow/crates/engine/src/exporter.rs

```diff
@@ -69,15 +90,173 @@ pub trait Exporter {
     /// This method should be cancellation safe and clean up any resources when dropped.
     async fn start(
         self: Box<Self>,
-        msg_chan: MessageChannel<Self::PData>,
-        effect_handler: EffectHandler<Self::PData>,
-    ) -> Result<(), Error<Self::PData>>;
+        msg_chan: MessageChannel<PData>,
+        effect_handler: EF,
+    ) -> Result<(), Error<PData>>;
+}
+
+/// Handles side effects for the exporter.
+///
+/// The `PData` type parameter represents the type of message the exporter will consume.
+///
+/// 2 implementations are provided:
+///
+/// - [`NotSendEffectHandler<PData>`]: For thread-local (!Send) exporters. Uses `Rc` internally.
+///   It's the default and preferred effect handler.
+/// - [`SendEffectHandler<PData>`]: For thread-safe (Send) exporters. Uses `Arc` internally and
+///   supports sending across thread boundaries.
+///
+/// Note for implementers: Effect handler implementations are designed to be cloned so the cost of
+/// cloning should be minimal.
+pub trait EffectHandlerTrait<PData> {
+    /// Returns the name of the exporter associated with this handler.
+    fn exporter_name(&self) -> &str;
+
+    // More methods will be added in the future as needed.
+}
+
+/// A `!Send` implementation of the EffectHandlerTrait.
+#[derive(Clone)]
+pub struct NotSendEffectHandler<PData> {
+    /// The name of the exporter.
+    exporter_name: Rc<str>,
+
+    /// A 0 size type used to parameterize the `EffectHandler` with the type of message the exporter
+    /// will consume.
+    _pd: PhantomData<PData>,
+}
+
+/// Implementation for the `!Send` effect handler.
+impl<PData> NotSendEffectHandler<PData> {
+    /// Creates a new local (!Send) `EffectHandler` with the given exporter name.
+    /// This is the default and preferred effect handler for this project.
+    ///
+    /// Use this constructor when your exporter doesn't need to be sent across threads or
```

Comment:
Are we supporting sending of exporter across threads with `SendEffectHandler`? I thought this was only components that are not `Send` and the exporter itself needing to be sent to a different thread.

Reply:
`SendEffectHandler` alone is not enough to make an exporter sendable. However the current framework allows exporter implementers to make their exporter sendable (see the async_trait(?Send) macro).